### PR TITLE
Re-enable some CI tests for other distros

### PR DESCRIPTION
**Pull request by stevenengler**
Re-enables CI tests for ubuntu:20.04, debian:10-slim, fedora:32, and centos:8. The 'findutils' package is for our cmake rust unit tests which require the 'find' tool. This PR also stops running linting tests with Python 2.7.

The tests do not currently complete on ubuntu:16.04 (shadow/shadow#864) or when using clang (shadow/shadow#865).

### CODE REVIEW

File: .github/workflows/build_shadow.yml [hunk 1]
```diff
@@ -22,24 +19,24 @@ jobs:
       options: '--shm-size="1g"'
     strategy:
       matrix:
-        container: ['ubuntu:18.04', 'centos:7'] #, 'ubuntu:16.04', 'ubuntu:20.04', 'debian:10-slim', 'fedora:32', 'centos:8']
-        cc: ['gcc'] #, 'clang']
+        container: ['ubuntu:18.04', 'ubuntu:20.04', 'debian:10-slim', 'fedora:32', 'centos:7', 'centos:8'] # 'ubuntu:16.04'
```
Comment (sporksmith) [inline .github/workflows/build_shadow.yml:22]: Could you add a comment with a link to the issue for ubuntu 16.04?

File: .github/workflows/build_shadow.yml [hunk 2]
```diff
@@ -22,24 +19,24 @@ jobs:
...
+        cc: ['gcc'] # 'clang'
```
Comment (sporksmith) [inline .github/workflows/build_shadow.yml:23]: Can clang be enabled too?

File: .github/workflows/build_shadow.yml (hunk 1 above)
Comment (stevenengler) [reply .github/workflows/build_shadow.yml:23]: Added

File: .github/workflows/build_shadow.yml (hunk 2 above)
Comment (stevenengler) [reply .github/workflows/build_shadow.yml:25]: It doesn't yet build with clang for some platforms.

**Pull request merged**